rocketchat-core/src/main/java/com/rocketchat/core/RestImpl.java

```diff
@@ -148,34 +225,71 @@ public void onFailure(Call call, IOException e) {
             @Override
             public void onResponse(Call call, Response response) throws IOException {
                 if (!response.isSuccessful()) {
-                    procressCallbackError(response, callback);
+                    processCallbackError(response, callback);
                     return;
                 }
 
                 try {
                     JSONObject json = new JSONObject(response.body().string());
-                    System.out.println("RESPONSE: " + json.toString());
-                    callback.onSuccess();
+                    logger.info("Response = " + json.toString());
+
+                    JSONArray filesJSONArray = json.getJSONArray("files");
+                    int length = filesJSONArray.length();
+                    List<Attachment> attachments = new ArrayList<>(length);
+                    for (int i = 0; i < length; ++i) {
+                        attachments.add(new Attachment(filesJSONArray.getJSONObject(i), baseUrl.url().toString()));
+                    }
+
+                    callback.onGetRoomFiles(json.optInt("total"), attachments);
                 } catch (JSONException e) {
-                    e.printStackTrace();
+                    callback.onError(new RocketChatInvalidResponseException(e.getMessage(), e));
                 }
             }
         });
     }
 
     /**
-     * Builds the request url as {baseUrl}/api/v1/{method}
+     * Returns the correspondent Rest API method name accordingly with the room type.
+     *
+     * @param roomType The type of the room.
+     * @return A Rest API method name accordingly with the room type.
+     * @see #requestUrl(HttpUrl, String)
+     */
+    private String getRestApiMethodNameByRoomType(BaseRoom.RoomType roomType, String method) {
+        switch (roomType) {
+            case PUBLIC:
+                return "channels." + method;
+            case PRIVATE:
+                return "groups." + method;
+            default:
+                return "dm." + method;
+        }
+    }
+
+    /**
+     * Builds the request URL as {baseUrl}/api/v1/{method}
+     *
+     * @param baseUrl The base URL.
+     * @param method The method name.
+     * @return A HttpUrl pointing to the REST API call.
```

Comment:
please update the return type, it now returns an `HttpUrl.Builder`

Reply:
Anyway, this is a private method, let me check how we can filter it from javadocs generation